app/controllers/application_controller.rb

```diff
@@ -1,7 +1,14 @@
 class ApplicationController < ActionController::Base
-    def page_title
-        @page_title ||= "Haxxor News"
-    end
-    
-    helper_method :page_title
-end
+  def page_title
+    @page_title ||= "Haxxor News"
+  end
+  
+  helper_method :page_title
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
```

Comment:
memoize this as well so that every call to `current_user` will fetch from a cached value and not from the database?